src/many-ledger/src/main.rs

```diff
@@ -90,6 +95,8 @@ fn main() {
     } else if persistent.exists() {
         // Initial state is ignored.
         state = None;
+    } else if snapshots.exists() {
+        state = None;
```

Comment:
This sounds wrong. Maybe overwrite the persistent storage with the latest snapshot if it exists. WDYT?